scripts/sync_repos.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash -eux
+#
+# Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+# Exceptions. See /LICENSE for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Sync directories in the main Carbon repository into dedicated child
+# repositories to better match repository-oriented installing and tooling.
+
+ORIGIN_DIR="$PWD"
+COMMIT_SHA="$(git rev-parse --short $GITHUB_SHA)"
+COMMIT_SUMMARY="Original $(git show -s --pretty=full "$COMMIT_SHA")
+
+$(git diff --summary "${COMMIT_SHA}^!")
+"
+
+# Setup global git configuration.
+GIT_USERNAME="CarbonInfraBot"
+git config --global user.email "carbon-external-infra@google.com"
+git config --global user.name "$GIT_USERNAME"
+
+declare -A MIRRORS
+MIRRORS["utils/vim"]="vim-carbon-lang"
+
+for dir in "${!MIRRORS[@]}"; do
+  SRC_DIR="$dir"
+  DEST_REPO="${MIRRORS[$SRC_DIR]}"
+  DEST_REPO_URL="https://$GIT_USERNAME:$API_TOKEN_GITHUB@github.com/carbon-language/$DEST_REPO.git"
+  DEST_CLONE_DIR="$(mktemp -d)"
+
+  git clone --single-branch "$DEST_REPO_URL" "$DEST_CLONE_DIR"
+  cd "$DEST_CLONE_DIR"
+
+  # Print out the destination repository.
+  ls -al
```

Comment:
I'm a bit confused by this; what's the goal of printing this? Won't `rsync -av` already print a lot?

This makes me wonder though: what does the output of this script look like? Is it easy to ready? It seems like it may be a bunch of text.